tf2onnx/onnx_opset/controlflow.py

```diff
@@ -362,20 +362,30 @@ def version_7(cls, ctx, node, **kwargs):
         # may be removed from output_names below
         output_names = node.output.copy()
 
-        # Make maximum_iterations int64 and replace -1(tf) with maxsize(onnx). If the const node has no other consumers,
-        # modify it in place. Otherwise, make a new const node and leave the original unchanged.
+        # Make maximum_iterations int64 and replace -1(tf) with maxsize(onnx). If the const node has no other
+        # consumers, modify it in place. Otherwise, make a new const node and leave the original unchanged.
+        # if maximum_iterations is not const,should add an cast node(cast to int64)
         maximum_iterations_name = node.input[1]
-        maximum_iterations = node.inputs[1].get_tensor_value()
-        if maximum_iterations == -1:
-            maximum_iterations = np.iinfo(np.int64).max
-        consumers = ctx.find_output_consumers(maximum_iterations_name)
-        external_consumers = [c for c in consumers if c != node and c.type != 'TensorListReserve']
-        if len(external_consumers) == 0:
-            ctx.remove_node(node.inputs[1].name)
+        cast_mark = False
+        if node.inputs[1].type != "Const":
+            cast_mark = True
+        if cast_mark:
```

Comment:
We prefer `is_const()` to checking if `node.type == "Const"` since `ConstV2` is also const.

```suggestion
        if node.inputs[1].is_const():
```

Reply:
With the removal of the lower if/else, `cast_mark` variable becomes unnecessary.